airflow/migrations/versions/0142_2_9_1_remove_idx_last_scheduling_decision_.py

```diff
@@ -0,0 +1,50 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Remove ``idx_last_scheduling_decision`` index on last_scheduling_decision in dag_run table
+
+Revision ID: bff083ad727d
+Revises: 677fdbb7fc54
+Create Date: 2024-04-26 12:58:00.594762
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = 'bff083ad727d'
+down_revision = '677fdbb7fc54'
+branch_labels = None
+depends_on = None
+airflow_version = "2.9.1"
+
+
+def upgrade():
+    """Apply Remove idx_last_scheduling_decision index on last_scheduling_decision in dag_run table"""
+    # This index may have been created in 2.0.0, but we've since removed it from migrations
+    from contextlib import suppress
+
+    with suppress(sa.exc.DatabaseError):  # mysql does not support drop if exists index
+        op.drop_index("idx_last_scheduling_decision", table_name="dag_run", if_exists=True)
+
+
+def downgrade():
+    """Unapply Remove idx_last_scheduling_decision index on last_scheduling_decision in dag_run table"""
+    pass
```

Comment:
+1

Reply:
Alright, let's reintroduce the index for the downgrade process. I don't anticipate any inconsistency in the database since we're not removing any columns or tables that would require re-adding during the downgrade. However, I'm considering the possibility that even a minor side-effect might warrant re-adding the index, just to be on the safe side.

Reply:
If we do add it again, it'll be in a new migration anyways :)